nntrainer/utils/base_properties.h

```diff
@@ -133,28 +160,44 @@ template <typename T> class Property {
    *
    * @return T value
    */
-  operator T &() { return value; }
+  operator T &() { return get(); }
 
   /**
    * @brief cast operator for property
    *
    * @return T value
    */
-  operator const T &() const { return value; }
+  operator const T &() const { return get(); }
 
   /**
    * @brief get the underlying data
    *
    * @return const T& data
    */
-  const T &get() const { return value; }
+  const T &get() const {
+    NNTR_THROW_IF(value == nullptr, std::invalid_argument)
+      << "Cannot get property, property is empty";
+    return *value;
+  }
 
   /**
    * @brief get the underlying data
    *
    * @return T& data
    */
-  T &get() { return value; }
+  T &get() {
+    NNTR_THROW_IF(value == nullptr, std::invalid_argument)
+      << "Cannot get property, property is empty";
+    return *value;
+  }
+
+  /**
+   * @brief check if property is empty
+   *
+   * @return true empty
+   * @return false not empty
+   */
+  bool empty() const { return value == nullptr; }
```

Comment:
```isEmpty()```?

Users may think this "empties data", not "checks if it's empty"

Reply:
yes, I'll change this :)

Reply:
Sorry, I found an issue prevent to change to `isEmpty()`

there is a template function insided nntrainer like below

```cpp
template<typename T>
void some_func(T& data){

  if(data.empty()) {
   /// do something
  }
}

TEST(can_compile_this) {
  Property<int> prop;
  std::vector<Property<int>> vector_of_prop

  some_func(prop); // prop.isEmpty() will fail
  some_func(vector_of_prop);
}
```

as stl has fixed semantics that *.empty() for testing emptiness. (arrays, vectors or others...)

Reply:
Ok.. it seems that STL is the culprit. keep using empty() for this.